src/server/scripts/Spells/spell_item.cpp

```diff
@@ -4092,6 +4092,62 @@ class spell_item_skyguard_blasting_charges : public SpellScript
     }
 };
 
+// 23595 - Luffa
+class spell_item_luffa : public SpellScript
+{
+    PrepareSpellScript(spell_item_luffa);
+    std::list<uint32> removeList;
+    std::list<ObjectGuid> casterList;
+
+    SpellCastResult CheckCast()
+    {
+        if (Unit* caster = GetCaster())
+        {
+            Unit::AuraApplicationMap const& auras = caster->GetAppliedAuras();
+            for (Unit::AuraApplicationMap::const_iterator itr = auras.begin(); itr != auras.end(); ++itr)
+            {
+                Aura const* aura = itr->second->GetBase();
+                if (aura->GetSpellInfo()->GetEffectMechanic(EFFECT_0) != MECHANIC_BLEED || aura->GetCasterLevel() > 60 || aura->GetSpellInfo()->IsPositive())
+                    continue;
+
+                removeList.push_back(aura->GetId());
+                casterList.push_back(aura->GetCasterGUID());
+            }
+        }
+
+        if (removeList.empty() || casterList.empty())
+            return SPELL_FAILED_NOTHING_TO_DISPEL;
+        else
+            return SPELL_CAST_OK;
+    }
+
+    void HandleEffect(SpellEffIndex effIndex)
+    {
+        PreventHitDefaultEffect(effIndex);
```

Comment:
this prevents the default [`void Spell::EffectDispelMechanic(SpellEffIndex effIndex)`](https://github.com/avarishd/azerothcore-wotlk/blob/5a13809e8e9f491da95e13e47bab29bc718dd68d/src/server/game/Spells/SpellEffects.cpp#L5131)

maybe copy the above and adjust for this specific case:

OwnedAuras -> `GetAppliedAuras()`
dispel_list -> no need to store. RemoveAura and return